src/AspNet.Security.OAuth.SuperOffice/Implementation/DefaultSuperOfficeConfigurationManager.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace AspNet.Security.OAuth.SuperOffice.Implementation
+{
+    internal sealed class DefaultSuperOfficeConfigurationManager : SuperOfficeAuthenticationConfigurationManager
```

Comment:
This looks like a copy-paste of code from the Apple provider, which by extension would be inheriting this design issue: #396

Reply:
The design issue is that it caches the token globally, which means that the Apple provider (and by extension this provider) does not support being used for more than one hostname per instance of the application.